framework/src/media/audio/audio_manager.c

```diff
@@ -321,6 +355,44 @@ static audio_manager_result_t get_supported_sample_rate(int fd, uint32_t *sample
 	return AUDIO_MANAGER_SUCCESS;
 }
 
+static audio_manager_result_t audio_manager_mutex_lock(audio_card_type_t type)
```

Comment:
As per we decided init, many things should be changed.
Definitely, role of init is initialize card info including mutex.
That means, if g_actual_audio_in/out_card_id initialized, we finished finding card already
(Currently below code is in while loop, I don't remember when it changed....it should be located out of while loop)
```
/* update actual card here */
if (card_type == INPUT) {
	g_actual_audio_in_card_id = get_actual_audio_in_card_id();
} else {
	g_actual_audio_out_card_id = get_actual_audio_out_card_id();
}

```
So, now it is very simple that we just check g_audio_in/out_card_id at the start of each API of audio manager which is called from player & recorder (ex. set_audio_volume, set_audio_stream_in/out...)

Reply:
The code for assigning g_actual_xx_card_id is moved out of the while loop.